libs/SalesforceSwiftSDK/SalesforceSwiftSDK/Extensions/SFSmartSyncExtensions.swift

```diff
@@ -0,0 +1,260 @@
+            }
+        }
+
+        /**
+         Sync Down
+         
+         ```
+         syncManager.Promises.deleteSync(id:  id)
+         .then {
+         ..
+         }
+         ```
+         - parameters
+             - target: SFSyncDownTarget target
+             - soupName: Soup Name
+             - updateBlock: Block to invoke
+         - Returns: SFSmartStore wrapped in a promise.
+         */
+        func syncDown(target: SFSyncDownTarget, soupName: String, updateBlock: SFSyncSyncManagerUpdateBlock) -> Promise<SFSyncState> {
+            return Promise(.pending) {  resolver in
+                resolver.fulfill(self.api!.syncDown(with: target, soupName: soupName, update: updateBlock))
+            }
+        }
+        
+        /**
+         Sync Down
+         
+         ```
+         syncManager.Promises.deleteSync(id:  id)
+         .then {
+         ..
+         }
+         ```
+         - parameters
+             - target: SFSyncDownTarget target
+             - options: SFSyncOptions
+             - soupName: Soup Name
+             - updateBlock: Block to invoke
+         - Returns: SFSmartStore wrapped in a promise.
+         */
+        func syncDown(target: SFSyncDownTarget, options: SFSyncOptions, soupName: String, updateBlock: SFSyncSyncManagerUpdateBlock) -> Promise<SFSyncState> {
+            return Promise(.pending) {  resolver in
+                resolver.fulfill(self.api!.syncDown(with: target,options: options, soupName: soupName, update: updateBlock))
+            }
+        }
+
+        /**
+         Sync Down
+         
+         ```
+         syncManager.Promises.deleteSync(id:  id)
+         .then {
+         ..
+         }
+         ```
+         - parameters
+         - target: SFSyncDownTarget target
+         - options: SFSyncOptions
+         - soupName: Soup Name
+         - updateBlock: Block to invoke
+         - Returns: SFSmartStore wrapped in a promise.
+         */
+        func syncDown(target: SFSyncDownTarget, options: SFSyncOptions, soupName: String, syncName: String?, updateBlock: SFSyncSyncManagerUpdateBlock) -> Promise<SFSyncState> {
+            return Promise(.pending) {  resolver in
+                resolver.fulfill(self.api!.syncDown(with: target,options: options, soupName: soupName,syncName: syncName, update: updateBlock))
+            }
+        }
+        
+        /**
+         ReSync
+         
+         ```
+         syncManager.Promises.reSync(syncId: syncId, updateBlock: {})
+         .then {
+         ..
+         }
+         ```
+         - parameters
+         - syncId: NSNumber
+         - updateBlock: Block to invoke
+         - Returns: SFSyncState wrapped in a promise.
+         */
+
+        func reSync(syncId: NSNumber, updateBlock: SFSyncSyncManagerUpdateBlock) -> Promise<SFSyncState> {
+            return Promise(.pending) {  resolver in
+                resolver.fulfill(self.api!.reSync(syncId, update: updateBlock)))
```

Comment:
The reSync call returns right away - should the promise resolve when the reSync actually completes.
So we could do something similar to what we did in the react native bridge: https://github.com/forcedotcom/SalesforceMobileSDK-iOS/blob/master/libs/SalesforceReact/SalesforceReact/Classes/SFSmartSyncReactBridge.m#L107